pkg/action/validate.go

```diff
@@ -42,7 +66,20 @@ func existingResourceConflict(resources kube.ResourceList) error {
 			return errors.Wrap(err, "could not get information about the resource")
 		}
 
-		return fmt.Errorf("existing resource conflict: namespace: %s, name: %s, existing_kind: %s, new_kind: %s", info.Namespace, info.Name, existing.GetObjectKind().GroupVersionKind(), info.Mapping.GroupVersionKind)
+		// Allow adoption of the resource if it already has app.kubernetes.io/instance and app.kubernetes.io/managed-by labels.
+		lbls, err := accessor.Labels(existing)
+		if err == nil {
+			set := labels.Set(lbls)
+			if sel.Matches(set) {
+				requireUpdate = append(requireUpdate, info)
```

Comment:
Here we return the `resource.Info` for the _latest_ version of the chart, not the resource that we've retrieved from the cluster. This might cause changes to not be applied when computing the 3-way diff; this still needs to be verified.